spec/lib/appsignal/hooks/http_spec.rb

```diff
@@ -0,0 +1,50 @@
+describe Appsignal::Hooks::HttpHook do
+  before :context do
+    start_agent
+  end
+
+  context "with HTTP instrumentation enabled" do
+    describe "#dependencies_present?" do
+      subject { described_class.new.dependencies_present? }
+
+      it { is_expected.to be_truthy }
+    end
+
+    it "should instrument a http request" do
+      Appsignal::Transaction.create("uuid", Appsignal::Transaction::HTTP_REQUEST, "test")
+      expect(Appsignal::Transaction.current).to receive(:start_event)
+        .at_least(:once)
+      expect(Appsignal::Transaction.current).to receive(:finish_event)
+        .at_least(:once)
+        .with("request.http_rb", "GET http://www.google.com", nil, 0)
+
+      stub_request(:any, "http://www.google.com/")
+
+      HTTP.get("http://www.google.com")
+    end
+
+    it "should instrument a https request" do
+      Appsignal::Transaction.create("uuid", Appsignal::Transaction::HTTP_REQUEST, "test")
+      expect(Appsignal::Transaction.current).to receive(:start_event)
+        .at_least(:once)
+      expect(Appsignal::Transaction.current).to receive(:finish_event)
+        .at_least(:once)
+        .with("request.http_rb", "GET https://www.google.com", nil, 0)
+
+      stub_request(:any, "https://www.google.com/")
+
+      HTTP.get("https://www.google.com")
```

Comment:
This is the old way of testing this. We like to avoid having to assert method calls to determine if the right data is set on the transaction.

The [que specs](https://github.com/appsignal/appsignal-ruby/blob/220f83efd98db7102a3a38f12f1dc939e0dfa080/spec/lib/appsignal/integrations/que_spec.rb#L44-L89) are an example of how we'd like to test things now. You can update it to use that method or we'll update it later.